src/timer.rs

```diff
@@ -347,6 +356,37 @@ macro_rules! hal {
                     }
                 }
 
+                /// Restarts the timer in count down mode with user-defined prescaler and auto-reload register
+                pub fn restart_raw(&mut self, psc: u16, arr: u16)
+                {
+                    // pause
+                    self.tim.cr1.modify(|_, w| w.cen().clear_bit());
+
+                    self.tim.psc.write(|w| w.psc().bits(psc) );
+                    self.tim.arr.write(|w| w.arr().bits(arr) );
```

Comment:
From the normal `start` function, add `unsafe` and a `#[allow(unused_unsafe)]` attribute to silence the warnings.